lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -820,14 +820,23 @@ func (d *powerflex) deleteNVMeHost() error {
 	return client.deleteHost(host.ID)
 }
 
-// mapNVMeVolume maps the given volume onto this host.
-func (d *powerflex) mapNVMeVolume(vol Volume) (revert.Hook, error) {
-	revert := revert.New()
-	defer revert.Fail()
+// mapVolume maps the given volume onto this host.
+func (d *powerflex) mapVolume(vol Volume) (revert.Hook, error) {
+	var hostID string
 
-	hostID, err := d.createNVMeHost()
-	if err != nil {
-		return nil, err
+	switch d.config["powerflex.mode"] {
+	case "nvme":
+		unlock, err := locking.Lock(d.state.ShutdownCtx, "nvme")
+		if err != nil {
+			return nil, err
+		}
+
+		defer unlock()
+
+		hostID, err = d.createNVMeHost()
```

Comment:
Does this need a revert to cleanup on error?

Reply:
Pls see comment above.